web/sfe/index.ts

```diff
@@ -0,0 +1,480 @@
+import { fromByteArray } from "base64-js";
+import $ from "jquery";
+
+interface GlobalAuthentik {
+    brand: {
+        branding_logo: string;
+    };
+}
+
+function ak(): GlobalAuthentik {
+    return (
+        window as unknown as {
+            authentik: GlobalAuthentik;
+        }
+    ).authentik;
+}
+
+class SimpleFlowExecutor {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    challenge: any;
+    flowSlug: string;
+    container: HTMLDivElement;
+
+    constructor(container: HTMLDivElement) {
+        this.flowSlug = window.location.pathname.split("/")[3];
+        this.container = container;
+    }
+
+    get apiURL() {
+        return `/api/v3/flows/executor/${this.flowSlug}/?query=${encodeURIComponent(window.location.search.substring(1))}`;
+    }
+
+    start() {
+        $.ajax({
+            type: "GET",
+            url: this.apiURL,
+            success: (data) => {
+                this.challenge = data;
+                this.renderChallenge();
+            },
+        });
+    }
+
+    submit(data: { [key: string]: unknown } | FormData) {
+        $("button[type=submit]").addClass("disabled")
+            .html(`<span class="spinner-border spinner-border-sm" aria-hidden="true"></span>
+                <span role="status">Loading...</span>`);
+        let finalData: { [key: string]: unknown } = {};
+        if (data instanceof FormData) {
+            finalData = {};
+            data.forEach((value, key) => {
+                finalData[key] = value;
+            });
```

Comment:
Problem: I don't have MSEdge 18, only MSEdge 17.  After filling in the "Email / Username" field, I received the error from MSEdge that `SCRIPT438: [FormData] Object doesn't support property or method 'forEach'`; I tried a variety of options (`Array.from`, `for` loop, etc) and none of them resolved to show me the data. Not sure what's going on there. [I am informed](https://stackoverflow.com/questions/37938955/iterating-through-formdata-in-ie) that earlier versions of MSEdge _should_ support it, but that some of the polyfills are unreliable.